### PR TITLE
feat(cli): agent-grade tester lookup, truncation warnings, and output contract

### PR DESCRIPTION
**Pull request by rudrankriyam**
## Why

Follow-up to #2051/#2052: a real automation session surfaced five ergonomics gaps that make the CLI easy to misuse from scripts and agents. This PR ships all five.

## 1. Truncation warning on unpaginated lists

Unpaginated `list` commands print one page with exit 0 and no signal that more pages exist — a script grepping `.data` silently acts on partial results (observed: `testers list` returning 50 of 198). `printOutput`/`printOutputWithRenderers` now emit a stderr warning whenever a printed response still carries `links.next`:

- `Warning: showing 50 of 198 results; pass --paginate for all (or --next for the next page)` (count+total when paging metadata is present; count-only and generic fallbacks otherwise)
- Correctness hinges on an existing invariant: `PaginateAll` erases `links.next` on aggregates, so `Next != ""` at print time is exactly "single page, unpaginated". No flag plumbing; covers all ~243 paginate-capable commands via the shared printer.
- Bonus fix: `beta-groups list --internal/--external` paginates fully then truncates client-side via `--limit` with no signal — now warns when it actually drops matches.
- Two cmdtest expectations updated: they asserted empty stderr on single-page fixtures specifically to prove no auto-pagination; the warning is now the expected signal there.

## 2. Tester name filters

`beta-testers list` gains `--first-name` / `--last-name` (exact-match `filter[firstName]`/`filter[lastName]`, OpenAPI-verified), following the existing `--email` plumbing exactly. "Find Lei Wang" is now one server-side call instead of fetch-all-and-grep.

## 3. Groups-per-tester discoverability

`testers groups list --id` already existed but appeared in no help example or doc — it now does (parent examples, view "See also", commands/testflight.mdx).

## 4. `--resolve-testers` on tester usage metrics

Metrics dimensions carry bare tester IDs. With `--resolve-testers`, the output gains an additive `testers` sidecar map (id → email/name/state/inviteType); `data[]` stays byte-identical. Resolution prefers the response's own `included` resources (previously discarded), then batch-fetches the rest via new `WithBetaTestersIDs` (`filter[id]`, chunks of 50, paginated, 500-ID cap with warning — never silent).

## 5. Output-shape contract

AGENTS.md now codifies the de-facto rule: reads print Apple's envelope unmodified; mutation receipts/computed results are exported camelCase structs in `internal/asc/output_*.go` with registered renderers. New `ListEnvelope[T]`/`ItemList[T]` generics provide an additive flattened form (`totalCount`/`hasMore` embed the envelope, existing `.data` consumers untouched) for future/experimental commands. Five outlier types normalized (dsyms, build wait, CSV export/import summaries, sync) — exported, renderer-registered, JSON byte-identical (fixture-tested).

## Tests

~30 new tests: query-builder + HTTP filter assertions, resolve-testers cmdtest (included-satisfied + batched fetch + flag-off byte-compat), warning unit tests (count+total/count-only/generic/aggregated/typed-nil/renderer-failure), `PageDataLen` edge cases, beta-groups truncation warning, ListEnvelope marshal shape + registry fallback, byte-compat fixtures for moved types.

## Notes for review

- Full gate green locally (docs check, website docs check, format, lint 0 issues, full short suite).
- `TestBuildsNextBuildNumberRefreshesDeadlinesAcrossHistoryPages` flaked once under full-suite load during a local gate run (context deadline on page 2), passed 3× in isolation; pre-existing sensitivity — its margins were already widened upstream in a7b4e549b. Not touched here.
- Implemented by three parallel agents with disjoint file ownership; integrated and gated as one change.

## Summary by CodeRabbit

* **New Features**
  * Added exact first-name, last-name, and tester ID filters for TestFlight beta tester listings.
  * Added the `testers groups list` command and expanded related documentation.
  * Added optional `--resolve-testers` support for beta tester usage metrics.
  * Added structured output for TestFlight sync, beta tester imports/exports, dSYM downloads, and build-wait results.
* **Bug Fixes**
  * Added warnings for truncated results caused by additional pages or limits.
  * Improved pagination metadata and list counts in JSON output.

### CODE REVIEW

File: internal/cli/shared/shared.go [hunk 1]
```diff
@@ -1155,20 +1162,58 @@ func printOutputWithRenderers(data any, format string, pretty bool, tableRendere
...
+	links := page.GetLinks()
+	if links == nil || links.Next == "" {
+		return
+	}
+
+	count, countOK := asc.PageDataLen(page)
+	if countOK {
+		if withMeta, hasMeta := page.(interface{ GetMeta() json.RawMessage }); hasMeta {
+			if total, totalOK := asc.ParsePagingTotalOK(withMeta.GetMeta()); totalOK {
+				fmt.Fprintf(os.Stderr, "Warning: showing %d of %d results; pass --paginate for all (or --next for the next page)\n", count, total)
```
[the remark this answers is not in the document]
Comment (rudrankriyam) [reply internal/cli/shared/shared.go]: Fixed in 8c8bd3106: the warning now says 'use --paginate or --next where supported' instead of asserting flags the command may not define. Plumbing per-command flag knowledge into the shared printer wasn't worth the coupling; giving the flagless list commands real pagination flags is a good follow-up.

File: internal/cli/testflight/metrics_beta_tester_usages.go [hunk 2]
```diff
@@ -15,11 +16,33 @@ import (
...
+// betaTesterUsageTesterInfo describes one resolved beta tester in the testers
+// sidecar keyed by tester ID.
+type betaTesterUsageTesterInfo struct {
+	ID         string `json:"id"`
+	Email      string `json:"email,omitempty"`
+	FirstName  string `json:"firstName,omitempty"`
+	LastName   string `json:"lastName,omitempty"`
+	State      string `json:"state,omitempty"`
+	InviteType string `json:"inviteType,omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (rudrankriyam) [reply internal/cli/testflight/metrics_beta_tester_usages.go]: Fixed in 8c8bd3106: BetaTesterUsagesPage and BetaTesterUsageTesterInfo now live in internal/asc/output_beta.go with a registered direct renderer — table mode gains a real metrics table (tester/period/sessions/crashes/feedback) plus the resolved-testers table when the sidecar is present. JSON output unchanged.

File: internal/cli/shared/shared.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rudrankriyam) [reply internal/cli/shared/shared.go]: Fixed in 8c8bd3106: the warning now says 'use --paginate or --next where supported' instead of asserting flags the command may not define. Plumbing per-command flag knowledge into the shared printer wasn't worth the coupling; giving the flagless list commands real pagination flags is a good follow-up.

File: internal/cli/testflight/metrics_beta_tester_usages.go [hunk 3]
```diff
@@ -184,6 +208,9 @@ func paginateBetaTesterUsages(ctx context.Context, client *asc.Client, appID str
 		}
 
 		combined.Data = append(combined.Data, parsed.Data...)
+		if len(combined.Included) == 0 && len(parsed.Included) > 0 {
+			combined.Included = parsed.Included
+		}
```
[the remark this answers is not in the document]
Comment (rudrankriyam) [reply internal/cli/testflight/metrics_beta_tester_usages.go]: Fixed in 635b8e0b6 — included arrays now merge across all metric pages (unmarshal per page, re-marshal once), so multi-page resolution is satisfied from Apple's own resources before any filter[id] fetch. Regression test pins two pages with included testers resolving with exactly 2 requests.

File: internal/cli/shared/shared.go [hunk 4]
```diff
@@ -1155,20 +1162,58 @@ func printOutputWithRenderers(data any, format string, pretty bool, tableRendere
...
+	return nil
+}
+
+// warnMorePages emits a stderr notice when rendered output is a single page of
+// a larger collection. PaginateAll clears links.next on aggregated results, so
+// a non-empty next link at print time means the output is one unpaginated page.
+func warnMorePages(data any) {
+	page, ok := data.(asc.PaginatedResponse)
+	if !ok {
+		return
```
[the remark this answers is not in the document]
Comment (rudrankriyam) [reply internal/cli/shared/shared.go:1192]: Fixed in 635b8e0b6 — BetaTesterUsagesResponse, BetaBuildUsagesResponse, and BetaTesterUsagesPage now implement GetLinks/GetData via lazy parsing, so metrics prints participate in the truncation warning while the printed bytes stay raw and identical. cmdtest pins the exact warning for a single page with links.next.
